pkg/util/docker/containers.go

```diff
@@ -265,3 +271,35 @@ func (d *DockerUtil) cleanupCaches(containers []types.Container) {
 	}
 	d.Unlock()
 }
+
+var missingIP = net.ParseIP("0.0.0.0")
+
+func isMissingIP(addrs []containers.NetworkAddress) bool {
+	for _, addr := range addrs {
+		if addr.IP.Equal(missingIP) {
+			return true
+		}
+	}
+	return false
+}
+
+func correctMissingIPs(addrs []containers.NetworkAddress, hostIPs []string) []containers.NetworkAddress {
+	if len(hostIPs) == 0 {
+		return addrs // cannot detect host list, will return the addresses as is
+	}
+
+	var correctedAddrs []containers.NetworkAddress
```

Comment:
nit: it would be better to pre-allocate this slice with the final size

Reply:
actually, we don't know that in advance since hostIPs is not necessarily one 🤷‍♂